app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyHomeDetailScreen.kt

```diff
@@ -141,21 +149,31 @@ fun SurveyDetailStartScreen(survey: Survey, modifier: Modifier = Modifier) {
 }
 
 @Composable
-fun SurveyToolbar(modifier: Modifier, showBack: Boolean, showClose: Boolean) {
+fun SurveyToolbar(navigator: DestinationsNavigator, modifier: Modifier, showBack: Boolean, showClose: Boolean) {
     Box(modifier = modifier) {
         if (showBack) {
-            Image(
-                painter = painterResource(id = R.drawable.ic_back_accent),
-                contentDescription = stringResource(id = R.string.survey_detail_back_icon),
+            IconButton(
+                onClick = { navigator.popBackStack() },
```

Comment:
Since it appears we will use different destinations for each survey question, it looks like we really don't need the `HorizontalPager` in `SurveyHomeDetailScreen` 🤔